src/payment/payment.service.ts

```diff
@@ -1,26 +1,71 @@
-import { Injectable } from '@nestjs/common';
+import { HttpException, HttpStatus, Injectable, Req } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { User } from 'src/user/entities/user.entity';
+import { UserInterface } from 'src/user/interfaces/user.interface';
 import { CreatePaymentDto } from './dto/create-payment.dto';
 import { UpdatePaymentDto } from './dto/update-payment.dto';
+import { PaymentInterface } from './interfaces/payment.interface';
 
 @Injectable()
 export class PaymentService {
-  create(createPaymentDto: CreatePaymentDto) {
-    return 'This action adds a new payment';
+  constructor(
+    @InjectModel('Payment') private readonly paymentModel: Model<PaymentInterface>,
+    @InjectModel('User') private readonly userModel: Model<UserInterface>,
+  ) {}
+
+  async create(createPaymentDto: CreatePaymentDto) {
+    const user = await this.userModel.findById(createPaymentDto.user).exec();
+
+    if (user) {
+      const payment = await new this.paymentModel(createPaymentDto);
+      return payment.save();
+    } else {
+      throw new HttpException('User not found', HttpStatus.NOT_FOUND);
+    }
+  }
+
+  async findAll() {
+    const payment = await this.paymentModel.find().exec();
+    if (!payment || !payment[0]) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return payment;
   }
 
-  findAll() {
-    return `This action returns all payment`;
+  async findAllPaymentsForUser(userId: User) {
```

Comment:
wszędzie nazywaliśmy `findAllByUserID`, ta nazwa też jest jak najbardziej spoko, ale później przy dalszym developmencie  nie będzie problemu, że funkcję odpowiadające za to samo w różnych modelach różnie się nazywają ;)